NSubstitute.DbConnection/NSubstitute.DbConnection.Tests/QueryTests.cs

```diff
@@ -156,4 +156,38 @@ public void ResultsShouldDependOnParameters(int resultSetCount, int rowCount)
 
         callCount.Should().Be(resultSetCount);
     }
+
+    [Test]
+    public void ShouldThrow()
+    {
+        var mockConnection = Substitute.For<DbConnection>().SetupCommands();
+        mockConnection.SetupQuery("delete from table")
+            .Throws(new Exception());
+
+        using var command = mockConnection.CreateCommand();
+        command.CommandText = "delete from table";
+        mockConnection.Open();
+
+        Assert.Throws<Exception>(() => { command.ExecuteReader(); });
+    }
+
+    [TestCase(0)]
+    [TestCase(1)]
+    [TestCase(2)]
+    [TestCase(4)]
+    [TestCase(8)]
+    public void ShouldThrowDependOnParameters(int id)
+    {
+        var mockConnection = Substitute.For<DbConnection>().SetupCommands();
+        mockConnection.SetupQuery("delete from table where id=@id")
```

Comment:
Can you set up queries for all the test cases here each time a test case runs, with a different message in the exception for each? Then when you assert the exception is thrown, you can also assert on the message being correct, which will verify that the selection of result set is working correctly for the cases where exceptions are thrown

Reply:
Done but I skipped the comparison of exception message since it's enough to compare the exception object itself.